### PR TITLE
feat(OMN-9762): add normalize_event_bus normalization function

### PR DESCRIPTION
**Pull request by jonahgabriel**
## Summary

Adds `normalize_event_bus()` — a pure dict→dict transform that strips the legacy event_bus block and top-level topic list keys (`event_bus`, `subscribe_topics`, `publish_topics`, `topics`) so legacy contract YAML passes canonical `extra="forbid"` Pydantic validation.

Phase 2 of the corpus classification + normalization layer (parent epic **OMN-9757**, plan: `docs/plans/2026-04-25-corpus-classification-and-normalization-layer.md`, Task 5).

This is **compatibility scaffolding**, not durable canonical conversion — callers that need the event-bus declarations must capture them (e.g., into a typed `ModelEventBusConfig`) before invoking the normalizer.

This PR creates the `normalization/` package skeleton; subsequent Phase 2 tickets (OMN-9761/OMN-9763/OMN-9764/OMN-9765/OMN-9766) will append further family normalizers and the `compose_normalization_pipeline`. Coordinate with OMN-9761 — both create `contract_normalizer.py`; whichever lands second will reconcile.

## Linear

- Closes **OMN-9762** — [Task 5: Phase 2 — Normalization — normalize_event_bus](https://linear.app/omninode/issue/OMN-9762)
- Parent epic: OMN-9757
- `dod_evidence`: unit_test (command below)

## Changes

- `src/omnibase_core/normalization/__init__.py` — new package with module docstring
- `src/omnibase_core/normalization/contract_normalizer.py` — `normalize_event_bus()` + `_LEGACY_EVENT_BUS_KEYS` constant; uses canonical `JsonType` from `omnibase_core.types.type_json`
- `tests/unit/normalization/test_contract_normalizer.py` — 6 unit tests (strip behavior, idempotency, input non-mutation, preservation of non-event-bus fields)
- `tests/unit/normalization/__init__.py` — SPDX-headered

## Test plan

- [x] `uv run pytest tests/unit/normalization/test_contract_normalizer.py -v` → 6 passed
- [x] `uv run ruff format src/omnibase_core/normalization tests/unit/normalization` → clean
- [x] `uv run ruff check src/omnibase_core/normalization tests/unit/normalization` → clean
- [x] `uv run mypy src/omnibase_core/normalization --strict` → clean
- [x] `uv run pre-commit run --all-files` → all hooks pass
- [ ] CI green (sharded full-suite verification — see workflow runs)

Corpus context: 291 contract files affected by `family_legacy_event_bus`.

## Summary by CodeRabbit

* **New Features**
  * Introduced contract normalization layer to handle legacy contract data formats
  * Automatically removes deprecated event-bus and topic-list related fields during processing
  * Ensures legacy contracts proceed to strict canonical schema validation without conflicts

* **Tests**
  * Added comprehensive unit test coverage for contract normalization, including edge cases and idempotency

### CODE REVIEW

File: src/omnibase_core/normalization/contract_normalizer.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    Drops `event_bus`, `subscribe_topics`, `publish_topics`, and `topics` keys
+    from the raw contract dict so the remainder passes canonical
+    extra="forbid" validation. Returns a new dict; does not mutate the input.
+    Idempotent on dicts that already lack these keys.
+
+    The event-bus declarations are preserved separately by callers that need
+    them (e.g., into a typed ModelEventBusConfig); this function only removes
+    the keys that would otherwise fail validation.
+    """
+    return {k: v for k, v in raw.items() if k not in _LEGACY_EVENT_BUS_KEYS}
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnibase_core/normalization/contract_normalizer.py:74]: This finding is misclassified for the documented context. The function is not a runtime backward-compatibility shim — it is **migration scaffolding** for a one-shot corpus normalization pass, explicitly authorized by parent epic **OMN-9757** and the plan at `docs/plans/2026-04-25-corpus-classification-and-normalization-layer.md` (Phase 2).

The docstring at the top of `contract_normalizer.py` calls this out directly: *"Each function targets a single migration family and is COMPATIBILITY SCAFFOLDING — it removes or rewrites fields without preserving the dropped content."* The PR body, Linear ticket, and module docstring all spell this out as plan-authorized scaffolding.

The module is consumed at validation time by the corpus migration pipeline (Task 9: `compose_normalization_pipeline`), not at runtime in production hot paths. Once 380+ legacy contracts have been canonicalized in-place, this layer is removed (per Phase 4 of the plan). Resolving as not-applicable.

File: src/omnibase_core/normalization/contract_normalizer.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+Functions in this module:
+    - normalize_event_bus (OMN-9762, family_legacy_event_bus): strips the
+      legacy event_bus block and top-level topic list keys.
+"""
+
+from omnibase_core.types.type_json import JsonType
+
+_LEGACY_EVENT_BUS_KEYS = frozenset(
+    {"event_bus", "subscribe_topics", "publish_topics", "topics"}
+)
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnibase_core/normalization/contract_normalizer.py]: The Enum suggestion would be over-engineering here. These four strings (`event_bus`, `subscribe_topics`, `publish_topics`, `topics`) are an **internal lookup set** scoped to a single function — they are never crossed across a process boundary, never serialized over Kafka or HTTP, never exposed in any public API surface. They are the migration family's "drop-set" and only ever consulted by `_LEGACY_EVENT_BUS_KEYS` membership.

The "Enums for external contract surfaces" rule applies to topic names, status enums, kind discriminators — values that appear in YAML, on the wire, or in stored state. A frozenset of legacy-key sentinels in a normalization helper does not meet that bar; an Enum here adds 8 lines of indirection with zero callers benefiting. The plan's pseudocode itself uses a `frozenset` for the same reason.

Sibling normalizers in the same module (e.g., `strip_legacy_metadata` in OMN-9761, also from the plan) follow the identical `frozenset` pattern. Resolving as not-applicable.
